test/index.test.js

```diff
@@ -0,0 +1,145 @@
+  var floodlight;
+  var analytics;
+  var options = {
+    source: '654757884637545',
+    events: [
+      {
+        key: 'Watched Westworld',
+        value: {
+          event: 'Watched Westworld',
+          cat: 'activityTag',
+          type: 'groupTag',
+          customVariable: [
+            {
+              key: 'favoriteCharacter',
+              value: 'u1'
+            },
+            {
+              key: 'episode',
+              value: 'u2'
+            }
+          ]
+        }
+      },
+      {
+        key: 'Goodbye Pablo',
+        value: {
+          event: 'Goodbye Pablo',
+          cat: 'activityTag',
+          type: 'groupTag',
+          customVariable: []
+        }
+      }
+    ]
+  };
+
+
+  beforeEach(function() {
+    analytics = new Analytics();
+    floodlight = new Floodlight(options);
+    analytics.use(Floodlight);
+    analytics.use(tester);
+    analytics.add(floodlight);
+  });
+
+  afterEach(function() {
+    analytics.restore();
+    analytics.reset();
+    floodlight.reset();
+    sandbox();
+  });
+
+  it('should have the correct settings', function() {
+    analytics.compare(Floodlight, integration('DoubleClick Floodlight')
+      .option('source', '')
+      .mapping('events'));
+  });
+
+  describe('after loading', function() {
+    beforeEach(function(done) {
+      analytics.once('ready', done);
+      analytics.initialize();
+    });
+
+    describe('track', function() {
+      beforeEach(function() {
+        analytics.spy(floodlight, 'load');
+      });
+
+      before(function() {
+        // stubbing cachebuster logic
+        sinon.stub(Math, 'random').returns(0.27005030284556764);
+      });
+
+      it('should fire a basic floodlight tag properly', function() {
+        var imgTag = '<img src="http://ad.doubleclick.net/activity'
+          + ';src=' + options.source
+          + ';type=' + options.events[1].value.type
+          + ';cat=' + options.events[1].value.cat
+          + ';ord=2700503028455676400?">';
+
+        analytics.track('Goodbye Pablo');
+        analytics.called(floodlight.load);
+        analytics.loaded(imgTag);
+      });
+
+      it('should fire a floodlight tag with custom variables properly', function() {
+        var imgTag = '<img src="http://ad.doubleclick.net/activity'
+          + ';src=' + options.source
+          + ';type=' + options.events[0].value.type
+          + ';cat=' + options.events[0].value.cat
+          + ';ord=2700503028455676400'
+          + ';u1=Maeve'
+          + ';u2=4?">';
+
+        analytics.track('Watched Westworld', {
+          favoriteCharacter: 'Maeve',
+          episode: '4'
+        });
+        analytics.called(floodlight.load);
+        analytics.loaded(imgTag);
+      });
+
+      describe('noops', function() {
+        it('should noop if no mapped tags are found for an event', function() {
+          analytics.track('They should announce a sequel to Groudhog Day and then just rerelease the original');
```

Comment:
lmaoooooo arguably the best catch of the already incredible code review thus far @jgershen @sperand-io